clash-prelude/src/Clash/Sized/Vector.hs

```diff
@@ -1492,7 +1492,7 @@ iterate SNat = iterateI
 -- \"'iterateI' @f z@\" corresponds to the following circuit layout:
 --
 -- <<doc/iterate.svg>>
-iterateI :: KnownNat n => (a -> a) -> a -> Vec n a
+iterateI :: forall n a. KnownNat n => (a -> a) -> a -> Vec n a
```

Comment:
This should be logged in the Changelog, as I think you had to type-apply them in reverse order before this change. I assume that's why you changed the order in the first place.

Reply:
Oh, no I just added it to make the order explicit. The order shouldn't have changed, but I'll test!

Reply:
This works in Clash 1.2.1:

```
Clash.Prelude> iterateI @5 @Int succ 3
<3,4,5,6,7>
```

I'd like to keep the explicit order in though.

Reply:
Ah, my bad. I thought it was type variables on the RHS of `=>` are traverse before they're traversed on the `LHS`, and then you'd encounter `a` before `n`.